### ISSUE TITLE
this._super is undefined in "data" option function

### ISSUE THREAD

**Issue by flyfoxuk** [ractivejs/ractive#2918]
## Description:

this._super is not defined when attempting to call it in the exact example given in the docs page "Advanced configuration", Data section example.  Suspect _super is not defined when used with a data option function, despite what the doc example says?

## Versions affected:
Latest: 0.8.12

## Platforms affected:
Safari on macOS, expect all others too.

## Reproduction:
https://jsfiddle.net/8yvk5cp8/
Open browser console, see error as follows:
TypeError: this._super is not a function. (In 'this._super(_data)', 'this._super' is undefined)

## Code:

```
var Component1 = Ractive.extend({
    data: {
        formatTitle: function (title) {
            return '"' + title.toUpperCase() + '"';
        }
    }
});

var Component2 = Component1.extend({
    data: function( data ) {
        this._super( data );
        data.scale = 5;
    }
});

var ractive = new Component2({
    data: { foo: 'bar' }
})
```

**Comment by ghost** [comment]
TL;DR: It's a docs issue 🙂 and `this._super` is no longer needed in function data.

After a bit of digging, I found https://github.com/ractivejs/ractive/issues/1413 and https://github.com/ractivejs/ractive/pull/1594 (see Initialisation functions in the PR description). In a gist, the behavior changed between 0.6 and 0.7 (there was an overhaul in data handling here), was removed from the [0.7 docs](http://docs.ractivejs.org/0.7/advanced-configuration) but came back in the [0.8 docs](http://docs.ractivejs.org/0.8/advanced-configuration). Data is [no longer supplied to function data](https://github.com/ractivejs/ractive/blob/17e74e20b46dabb3f1342d70e254b1fb438442d3/src/Ractive/config/custom/data.js#L108) and `this._super(data)` on function data is no longer needed. You'll only need to return an object containing your augments/overrides for the child component. Ractive will then [shallow-copy the parent data to the child data](https://github.com/ractivejs/ractive/blob/17e74e20b46dabb3f1342d70e254b1fb438442d3/src/Ractive/config/custom/data.js#L123), with child data taking precedence in the event of collisions.

[Here's an example](https://jsfiddle.net/0a6bL9zu/).

```js
var Component1 = Ractive.extend({
    data: () => ({
        formatTitle: function (title) {
            return '"' + title.toUpperCase() + '"';
        }
    })
});

var Component2 = Component1.extend({
    data: () => ({
    	scale: 5
    })
});

var ractive = new Component2({
    el: '#fixture',
    template: '{{ foo }} {{ scale }} {{ formatTitle(foo) }}',
    data: { foo: 'bar' }
});
```

[It also works on plain object data](https://jsfiddle.net/3xozgwgj/) (which is normally discouraged, but if you know that the data will be attached to the prototype instead of the instance, then you're good).

---

The docs are being [revised](https://github.com/ractivejs/ractivejs.github.io) at the moment. Working prototype can be found at [ractive.js.org](https://ractive.js.org/). Thank you for letting us know about this issue. 😄

**Comment by ghost** [comment]
Data merging explained better at https://ractive.js.org/api/initialization-options/#data
